src/rgw/driver/rados/rgw_sal_rados.cc

```diff
@@ -2286,7 +2322,7 @@ int RadosObject::read_attrs(const DoutPrefixProvider* dpp, RGWRados::Object::Rea
   return read_op.prepare(y, dpp);
 }
 
-int RadosObject::set_obj_attrs(const DoutPrefixProvider* dpp, Attrs* setattrs, Attrs* delattrs, optional_yield y)
+int RadosObject::set_obj_attrs(const DoutPrefixProvider* dpp, Attrs* setattrs, Attrs* delattrs, optional_yield y, uint32_t flags)
```

Comment:
here in line [2330](https://github.com/ceph/ceph/pull/57060/files#diff-8d5ab77cf6b24e95c2c7b352c52d1e254781f6c9e7fb3fd148bddc8a3f4321e7R2330) we adjust mtime to trigger replication on a modified object. now, since we don't want that to happen in case of `handle_replication_status_header()`, you are doing the right thing by passing in value 0 for `flags`. we just need to change the code here to modify mtime only based on the condition that `FLAG_LOG_OP` is set and not otherwise.

Reply:
Updated logic to only increment mtime when `FLAG_LOG_OP` is set. Please review updated logic